### PR TITLE
network: make GRE and GRETAP support Key= or friends

### LINKED ISSUE #12144: systemd-networkd complete passing tunnel options
**Is your feature request related to a problem? Please describe.**
Currently the netdev [Tunnel] options have `Key=`, `InputKey=` and `OutputKey=` parameters that were originally added for VTI/VTI6 tunnels, but are just as applicable to any GRE variant. In fact ERSPAN already does use those even though the documentation has not been updated reflect that fact.

ERSPAN also added `SerializeTunneledPackets=` that is also applicable to all GRE variants as well.

**Describe the solution you'd like**
As far as I understand the exact parameter passing logic for `key`, `ikey`, `okey` and `erspan_sequence` as is now in `netdev_erspan_fill_message_create` (src/network/netdev/tunnel.c) can be coped to `netdev_ip6gre_fill_message_create` and `netdev_gre_fill_message_create` as is.

The only issue would be the somewhat unfortunate variable naming with `erspan_sequence` and code repetition.

Sync the documentation. At the very least to reflect that ERSPAN can already use the Key*= parameters (not just VTI/VTI6).

### PR DESCRIPTION
**Pull request by yuwata**
Closes #12144.

### CODE REVIEW

**Comment by kristjanbb** [comment]
Looks good. Thank you.

File: src/network/netdev/tunnel.c [hunk 1]
```diff
@@ -854,9 +840,9 @@ const NetDevVTable ip6tnl_vtable = {
 
 const NetDevVTable erspan_vtable = {
         .object_size = sizeof(Tunnel),
-        .init = erspan_init,
+        .init = gre_init,
```
Comment (ssahani) [inline src/network/netdev/tunnel.c]: Please make this something like ```gre_erspan_init``` . They are sharing the same func

File: src/network/netdev/tunnel.c [hunk 2]
```diff
@@ -854,9 +840,9 @@ const NetDevVTable ip6tnl_vtable = {
...
         .sections = "Match\0NetDev\0Tunnel\0",
-        .fill_message_create = netdev_erspan_fill_message_create,
-        .create_type = NETDEV_CREATE_INDEPENDENT,
+        .fill_message_create = netdev_gre_fill_message_create,
```
Comment (ssahani) [inline src/network/netdev/tunnel.c]: same here

File: src/network/netdev/tunnel.h [hunk 3]
```diff
@@ -29,7 +29,7 @@ typedef struct Tunnel {
         int family;
         int ipv6_flowlabel;
         int allow_localremote;
-        int erspan_sequence;
+        int gre_sequence;
         int isatap;
```
Comment (ssahani) [inline src/network/netdev/tunnel.h:33]: same here

**Comment by yuwata** [comment]
@ssahani Thank you for the comments. I've renamed the variable and functions. PTAL.

**Comment by ssahani** [comment]
LGTM

**Pull request merged**